packages/calcite-components/src/assets/styles/_x-button.scss

```diff
@@ -0,0 +1,59 @@
+@mixin xButton(
+  $textColor: "--calcite-color-text-3",
+  $backgroundColor: "--calcite-color-foreground-2",
+  $hoverTextColor: "--calcite-color-text-3",
+  $hoverBackgroundColor: "--calcite-color-foreground-3",
+  $activeBorderColor: "--calcite-color-brand",
+  $borderWidth: "--calcite-border-width-md"
+) {
+  .x-button {
+    appearance: none;
+    align-content: center;
+    cursor: pointer;
+    display: flex;
+    outline-color: transparent;
+    align-items: center;
+    justify-content: center;
+    margin: 0px;
+    align-self: center;
+    color: var(#{$textColor}, var(--calcite-color-text-3));
+    transition:
```

Comment:
Could you explicitly list the props that will be transitioned by state? Looks like it's `color` and `background-color`.